src/ImageSharp/Formats/Webp/Lossless/Vp8LEncoder.cs

```diff
@@ -76,6 +85,8 @@ internal class Vp8LEncoder : IDisposable
 
         private const int PaletteInvSize = 1 << PaletteInvSizeBits;
 
+        private static readonly byte[] Order = { 1, 2, 0, 3 };
```

Comment:
```suggestion
        // This uses C#'s compiler optimization to refer to assembly's static data directly.
        private static ReadOnlySpan<byte> Order => new byte[] { 1, 2, 0, 3 };
```

Reply:
@gfoidl would you think the same trick could be applied to the BitMask in `Vp8LBitReader`?

Reply:
You mean https://github.com/SixLabors/ImageSharp/blob/d0212221622b015706ca4d0728c269ec3caa5c7a/src/ImageSharp/Formats/Webp/BitReader/Vp8LBitReader.cs#L31-L40 ?

Then this C# optimization won't work, as it only works for single byte-size types (`byte`, `sbyte`, `bool`) due to endianess concerns. There is https://github.com/dotnet/runtime/issues/60948 to allow other (primitive and simple) types too.

One could type it a sequence of bytes in order to the ROS<byte>-trick, then cast that span to a ROS<uint> one. In that case a) code becomes ugly and b) same endianess-trouble occurs. I wouldn't do this -- once the aforementioned runtime-issue is resolved we may do a pass over the codebase to see where to apply this trick for other types than single byte-sized ones.

---

But this field should be made `static readonly` so that the JIT can treat it as "constant" (and for Tier-1 can skip the beforefield init check).

Reply:
yes that's what I meant. Thanks for the clarification.

Another question i have: For what sizes of array's would you recommend using this trick. I guess it should be rather small array's only? We have alot of `byte`, `sbyte` lookups in `WebpLookupTables`. Would it make sense to apply this there also or should we not go to crazy with this?

Reply:
I'd do it for all static data of type byte, sbyte, bool -- no matter what size.

Look at this [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCphA3oaY6QA5QCWAbgIYAuM5OJKVhcAJhAB2AGwCepYDL4BtALqkAzgH0uUKFzkBeUhJgB3eYpirStUjgA0pLI+KOUjgKyOkjgOyOADkcATlIAXwBuBiZoxgB6ONIlAAY1NnVSUTYAM2yYKAkeRx4IUgALLg5+HlNSmEL2GAy2CVIeMv4ASQAZWNIE0gAVKDkS0jAKiQBzfhS1IwcNUvUuAFt+LgyuYAgq0i4JUQ0YfjKIc3VVnR5SACUIdVlW9NIQOABKPtZOXn4MQTuMDEAHlpDIAMrMA4AHgAQpYAHykSEHUgGJEmcwKZRqWzJRzOUiuUjuUheUg+Uj+UhBUihSKEMJAA==). You can see that for the array and for the ROS there's data in the static segment of the assembly. So the size of the DLL won't increase by using the ROS-trick extensively. 

The big advantage is that the ROS refers to the static data directly, no need for the extra array allocation (even if static initialized, so as singleton), and that with the ROS the data is truly read-only (for the array you can write to the elements even if it is declared as readonly, as this is just for the reference to the array itself).